Automattic/wp-calypso
# Section Loader: create a separate SECTION_LOADING_SET action for showing the loading indicator

The section loader shows a little animated indicator in the middle of the masterbar when the section loading takes more than 400ms:

<img width="175" alt="Screenshot 2020-09-30 at 09 50 11" src="https://user-images.githubusercontent.com/664258/94658886-bb793480-0303-11eb-8c08-2f6bb972989b.png">

The flag is in `state.ui.isLoading` and we sort of abuse the `SECTION_SET` action to change it.

This PR introduces a separate `SECTION_LOADING_SET` action to change the flag. It doesn't really have much in common with `SECTION_SET`. It affects different reducers and is read by different selectors. The only user of the `isSectionLoading` flag is the `LayoutLoader` component.

One small step to clean up and optimize the Redux state for sections and routes.

**How to test:**
Verify that the `LayoutLoader` shows up as expected. I needed to add some delay (`await new Promise(r => setTimeout(r, ms))`) to the code that loads the section module to trigger it.
